l1-contracts/src.ts/hyperchain-upgrade.ts

```diff
@@ -181,9 +233,70 @@ async function integrateEraIntoBridgehubAndUpgradeL2SystemContract(deployer: Dep
 }
 
 async function upgradeL2Bridge(deployer: Deployer) {
-  // upgrade L2 bridge contract, we do this directly via the L2
-  // set initializeChainGovernance in L1SharedBridge
-  deployer;
+  // todo deploy l2 bridge here
+  // L2_SHARED_BRIDGE_IMPLEMENTATION_BYTECODE
+  const l2BridgeImplementationAddress = ADDRESS_ONE; // todo
+
+  // upgrade from L1 governance. This has to come from governacne on L1.
+  const l2BridgeAbi = ["function initialize(address, address, bytes32, address)"];
+  const l2BridgeContract = new ethers.Contract(ADDRESS_ONE, l2BridgeAbi, deployer.deployWallet);
+  const l2Bridge = l2BridgeContract.interface; //L2_SHARED_BRIDGE_INTERFACE;
+  const l2BridgeCalldata = l2Bridge.encodeFunctionData("initialize", [
+    deployer.addresses.Bridges.SharedBridgeProxy,
+    deployer.addresses.Bridges.ERC20BridgeProxy,
+    hashL2Bytecode(BEACON_PROXY_BYTECODE),
+    applyL1ToL2Alias(deployer.addresses.Governance),
+  ]);
+  const bridgeProxy: ITransparentUpgradeableProxy = ITransparentUpgradeableProxyFactory.connect(
+    ADDRESS_ONE,
+    deployer.deployWallet
+  ); // we just need the interface, so wrong address
+
+  const l2ProxyCalldata = bridgeProxy.interface.encodeFunctionData("upgradeToAndCall", [
+    l2BridgeImplementationAddress,
+    l2BridgeCalldata,
+  ]);
+
+  const mailboxFacet = new Interface(hardhat.artifacts.readArtifactSync("MailboxFacet").abi);
+  const factoryDeps = [];
+  const mailboxCalldata = mailboxFacet.encodeFunctionData("requestL2Transaction", [
+    process.env.CONTRACTS_L2_SHARED_BRIDGE_ADDR,
+    0,
+    l2ProxyCalldata,
+    priorityTxMaxGasLimit,
+    REQUIRED_L2_GAS_PRICE_PER_PUBDATA,
+    factoryDeps,
+    deployer.deployWallet.address,
+  ]);
+  const gasPrice = await deployer.deployWallet.getGasPrice();
+  const requiredValueForL2Tx = await deployer
+    .bridgehubContract(deployer.deployWallet)
+    .l2TransactionBaseCost(deployer.chainId, gasPrice, priorityTxMaxGasLimit, REQUIRED_L2_GAS_PRICE_PER_PUBDATA); //"1000000000000000000";
+
+  await deployer.executeUpgrade(
+    deployer.addresses.StateTransition.DiamondProxy,
+    requiredValueForL2Tx.mul(10),
```

Comment:
be careful not to keep this `mul` on mainnet : )

Reply:
we will not use this on stage/testnet/ mainnet :)